frontend/src/views/Members.vue

```diff
@@ -411,36 +385,72 @@ export default {
         this.kubeconfigDialog = true
       }
     },
-    confirmDelete (username) {
-      const memberName = escape(displayName(username))
+    async onRemoveUser (name) {
+      const removalConfirmed = await this.confirmRemoveUser(name)
+      if (removalConfirmed) {
+        await this.deleteMember(name)
+        if (this.isCurrentUser(name) && !this.isAdmin) {
+          if (this.projectList.length > 0) {
+            const p1 = this.projectList[0]
+            this.$router.push({ name: 'ShootList', params: { namespace: p1.metadata.namespace } })
+          } else {
+            this.$router.push({ name: 'Home', params: { } })
+          }
+        }
+      }
+    },
+    confirmRemoveUser (name) {
+      const memberName = escape(displayName(name))
       const projectName = escape(this.projectDetails.projectName)
       let messageHtml
-      if (this.isCurrentUser(username)) {
+      if (this.isCurrentUser(name)) {
         messageHtml = `Do you want to remove <span class="red--text text--darken-2 font-weight-bold">yourself</span> from the project <i>${projectName}</i>?`
       } else {
-        messageHtml = `Do you want to delete the member <i>${memberName}</i> from the project <i>${projectName}</i>?`
+        messageHtml = `Do you want to remove the member <i>${memberName}</i> from the project <i>${projectName}</i>?`
       }
       return this.$refs.confirmDialog.waitForConfirmation({
-        confirmButtonText: 'Delete',
-        captionText: 'Confirm Member Deletion',
+        confirmButtonText: 'Remove Member',
+        captionText: 'Confirm Remove Member',
         messageHtml,
         dialogColor: 'red'
       })
     },
-    async onDelete (username) {
-      const deletionConfirmed = await this.confirmDelete(username)
+    async onDeleteServiceAccount (serviceAccountName) {
+      let deletionConfirmed
+      if (isForeignServiceAccount(serviceAccountName, this.namespace)) {
+        deletionConfirmed = await this.confirmRemoveForeignServiceAccount(serviceAccountName)
+      } else {
+        deletionConfirmed = await this.confirmDeleteServiceAccount(serviceAccountName)
+      }
       if (deletionConfirmed) {
-        await this.deleteMember(username)
-        if (this.isCurrentUser(username) && !this.isAdmin) {
-          if (this.projectList.length > 0) {
-            const p1 = this.projectList[0]
-            this.$router.push({ name: 'ShootList', params: { namespace: p1.metadata.namespace } })
-          } else {
-            this.$router.push({ name: 'Home', params: { } })
-          }
-        }
+        await this.deleteMember(serviceAccountName)
       }
     },
+    confirmRemoveForeignServiceAccount (name) {
+      const projectName = escape(this.projectDetails.projectName)
```

Comment:
We should not do html escape. There should be a safe way to create escaped html. Whether via a template engine or via a Vue component that renders this html.

Reply:
let's create an issue for this an tackle it in a separate issue

Reply:
Yes, I think this is something that has nothing to do with this PR in particular. Moreover, we should change this in other places, too. I'll create an issue.

Reply:
Issue created: https://github.com/gardener/dashboard/issues/835